RFC: Refactor of ZIUHF and new ZIHF2 driver

YakBizzarro opened the pull request:
HF2Li is a lockin similar to the UHFLI. Here I used the driver of the UHFLI as a common base, with two child classes to create a driver for each instrument.

Moreover, in the signal output section:
- removed some dead code
- the signal amplitude now honor the amplitude definition (Vpk, Vrms, dBm) both in getter and setter
- the amplitude and offset are now scaled according to the right range (not sure if needed on the UHF)

Still WIP, not to be pulled yet, I checked only the basic parameters (no AuxChannels, Scope or Sweeper) on the HF2. To be tested again on the UHF

@astafan8 @jenshnielsen @WilliamHPNielsen

YakBizzarro commented:
@astafan8 @jenshnielsen @WilliamHPNielsen some comment on this?

astafan8 commented on qcodes/instrument_drivers/ZI/private/ZILI_generic.py:
```diff
@@ -0,0 +1,2123 @@
+                                            0, 'trigslope'),
+                            get_cmd=partial(self._getter, 'scopes', 0,
+                                            0, 'trigslope'),
+     
```
snake_case please :)

astafan8 replied:
could you go thorugh all the TODO that you have in this PR and make sure that they are still relevant? if you are a kind of person who likes to leave them in the code, perhaps, could you make them more explicit sometimes (for example, on this line, what kind of validation you are talking about, and please try to avoid weak language (like "seems") - these notes only have value if they are 100% correct.

astafan8 commented:
by the way, the description says that it is still WIP - is it true or not? are you finished? (frankly speaking, i did not look at the code before because it was still WIP). Has it been tested on a real instrument? please, mention all of this in the description.

YakBizzarro commented:
I tested it on an HF2 and partially on an UHF. While WIP, I put it here to receive some feedback before finishing it, just to see if, in your opinion is the right approach, since it's quite a lot of work and testing. If it sound correct, I will polish it for a real PR

astafan8 commented:
aha, ok, could you then describe the approach? because perhaps i missed it. Or is the approach presented in the bullet points of the description?

YakBizzarro commented:
I just took the existing UHF driver, moved to a "generic" file (commit fedb5301aa7642e92507b2cac079d91a7aa9e05b) then I subclassed it to "new" UHF and HF2 with their differences (mostly commit 2c221b7b4f704c6fc379483fe5708b8c955d3e48 and 2a83d862430b8977fc4a0ca52f4503f6fc187a68). It's easier to look at these two last commits, since the show what I modified

astafan8 commented:
@YakBizzarro any chance that the work on this PR will continue?

astafan8 commented:
@YakBizzarro any chance that the work on this PR will continue?

astafan8 commented:
@YakBizzarro

The driver in this pull request will not be supported by QCoDeS developers, and, hence, is a **community driver**. Please close this pull-request and open a new one against [`qcodes_contrib_drivers`](https://github.com/QCoDeS/Qcodes_contrib_drivers) repository. QCoDeS developers will close this pull-request unconditionally **in 2 weeks**.

[Read about `qcodes_contrib_drivers` repository and python package here.](https://github.com/QCoDeS/Qcodes_contrib_drivers#qcodes-contrib-drivers)

astafan8 commented:
@YakBizzarro 

I'm closing this PR. Feel free to open this against [Qcodes_contrib_drivers](https://github.com/QCoDeS/Qcodes_contrib_drivers).
